frontend/src/panels/GamePanel.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState, useEffect } from 'react';
+import CellGrid from '../game/CellGrid';
+import { cellStates } from '../game/Cell';
+import { makeStyles } from '@material-ui/core/styles';
+import Backdrop from '@material-ui/core/Backdrop';
+import Button from "@material-ui/core/Button";
+import Card from "@material-ui/core/Card";
+import CardContent from "@material-ui/core/CardContent";
+import Typography from "@material-ui/core/Typography";
+import Grid from "@material-ui/core/Grid";
+
+const useStyles = makeStyles(theme => ({
+    continueButton: {
+        float: "right",
+        margin: "auto"
+    },
+    gameStateText: {
+        display: "block",
+        textAlign: "center",
+        fontSize: "20px"
+    }
+}));
+
+const players = {
+    HUMAN: "human",
+    COMPUTER: "computer"
+}
+
+class InvalidMoveError extends Error {};
+
+export default function GamePanel(props) { 
```

Comment:
I would create another class that is the gameboard class, just a regular js class. The game panel could just create a new gameboard with the rows and columns necessary. I.E. new Game(rowCount, columnCount).

Then all this logic can exist inside of the game class and the GamePanel would only be responsible for calling the game function. game.makeMove(player, row, column) and game.checkForWinner could just be a private function within the game.makeMove function which just returns if there is a winner or not.

That would make testing pretty simple also since for the GamePanel you only need to make sure you are calling the Game functions and responding correctly to the game.makeMove return value.

Then testing the game is just plain old JS and not coupled to react at all.